python/paddle/fluid/tests/unittests/testsuite.py

```diff
@@ -156,8 +156,26 @@ def create_var(block, name, np_list, var_proto):
     return var_dict
 
 
+def var_cast(block, input):
+    if input.dtype == core.VarDesc.VarType.FP32 or input.dtype == core.VarDesc.VarType.FP32:
+        return input
+    out = block.create_var(dtype="float32", shape=[1])
+    op = block.append_op(
+        inputs={"X": input},
+        outputs={"Out": out},
+        type='cast',
+        attrs={
+            'out_dtype': core.VarDesc.VarType.FP32,
+            'in_dtype': input.dtype
+        })
+    op.desc.infer_var_type(block.desc)
+    op.desc.infer_shape(block.desc)
+    return out
+
+
 def append_loss_ops(block, output_names):
     mean_inputs = list(map(block.var, output_names))
+    mean_inputs = [var_cast(block, x) for x in mean_inputs]
```

Comment:
why they must be fp32? Can you do the cast in your own op test?

Reply:
another pr fix this problem， remove test first